src/overlays/actors/ovl_Bg_Hidan_Curtain/z_bg_hidan_curtain.c

```diff
@@ -21,25 +45,201 @@ const ActorInit Bg_Hidan_Curtain_InitVars = {
+    Actor_SetHeight(&this->actor, 20.0f);
+    this->type = (thisx->params >> 0xC) & 0xF; // needed for matching
+    if (this->type > 6) {
+        // Type is not set
+        osSyncPrintf("Error : object のタイプが設定されていない(%s %d)(arg_data 0x%04x)\n", "../z_bg_hidan_curtain.c",
+                     352, this->actor.params);
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    this->sizeIndex = ((this->type == 2) || (this->type == 4)) ? 1 : 0;
+    hcParams = &sHCParams[this->sizeIndex];
+    this->treasureFlag = (thisx->params >> 6) & 0x3F;
+    this->actor.params &= 0x3F;
+
+    if ((this->actor.params < 0) || (this->actor.params > 0x3F)) {
+        // Save bit is not set
+        osSyncPrintf("Warning : object のセーブビットが設定されていない(%s %d)(arg_data 0x%04x)\n",
+                     "../z_bg_hidan_curtain.c", 373, this->actor.params);
+    }
+    Actor_SetScale(&this->actor, hcParams->scale);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->collider.dim.pos.x = this->actor.posRot.pos.x;
+    this->collider.dim.pos.y = this->actor.posRot.pos.y;
+    this->collider.dim.pos.z = this->actor.posRot.pos.z;
+    this->collider.dim.radius = hcParams->radius;
+    this->collider.dim.height = hcParams->height;
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+    func_80061ED4(&thisx->colChkInfo, NULL, &sCcInfoInit);
+    if (this->type == 0) {
+        this->actionFunc = BgHidanCurtain_WaitForClear;
+    } else {
+        this->actionFunc = BgHidanCurtain_WaitForSwitchOn;
+        if ((this->type == 4) || (this->type == 5)) {
+            this->actor.posRot.pos.y = this->actor.initPosRot.pos.y - hcParams->riseDist;
+        }
+    }
+    if (((this->type == 1) && Flags_GetTreasure(globalCtx, this->treasureFlag)) ||
+        (((this->type == 0) || (this->type == 6)) && Flags_GetClear(globalCtx, this->actor.room))) {
+        Actor_Kill(&this->actor);
+    }
+    this->texScroll = Math_Rand_ZeroOne() * 15.0f;
+}
+
+void BgHidanCurtain_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgHidanCurtain* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void BgHidanCurtain_WaitForSwitchOn(BgHidanCurtain* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, this->actor.params)) {
+        if (this->type == 1) {
+            this->actionFunc = BgHidanCurtain_WaitForCutscene;
+            func_800800F8(globalCtx, 3350, -99, &this->actor, 0);
+            this->timer = 50;
+        } else if (this->type == 3) {
+            this->actionFunc = BgHidanCurtain_WaitForCutscene;
+            func_800800F8(globalCtx, 3360, 60, &this->actor, 0);
+            this->timer = 30;
+        } else {
+            this->actionFunc = BgHidanCurtain_TurnOff;
+        }
+    }
+}
+
+void BgHidanCurtain_WaitForCutscene(BgHidanCurtain* this, GlobalContext* globalCtx) {
```

Comment:
I don't think that function should be named for what it is intended to do (wait for the cutscene to end) when all it does is wait on the timer like `WaitForTimer`
Maybe call this `WaitOffTimer` and rename `WaitForTimer` to `WaitOnTimer`? `TurnOffAfterTimer`/`TurnOnAfterTimer`?
Or just get rid of `Timer` in the name idk